packages/flutter/lib/src/rendering/box.dart

```diff
@@ -2360,42 +2558,105 @@ abstract class RenderBox extends RenderObject {
     }());
   }
 
-  bool _clearCachedData() {
-    if ((_cachedBaselines != null && _cachedBaselines!.isNotEmpty) ||
-        (_cachedIntrinsicDimensions != null && _cachedIntrinsicDimensions!.isNotEmpty) ||
-        (_cachedDryLayoutSizes != null && _cachedDryLayoutSizes!.isNotEmpty)) {
-      // If we have cached data, then someone must have used our data.
-      // Since the parent will shortly be marked dirty, we can forget that they
-      // used the baseline and/or intrinsic dimensions. If they use them again,
-      // then we'll fill the cache again, and if we get dirty again, we'll
-      // notify them again.
-      _cachedBaselines?.clear();
-      _cachedIntrinsicDimensions?.clear();
-      _cachedDryLayoutSizes?.clear();
+  void _debugVerifyDryBaselines() {
+    assert(() {
+      final List<DiagnosticsNode> messages = <DiagnosticsNode>[
+        ErrorDescription(
+          'The constraints used were $constraints.',
+        ),
+        ErrorHint(
+          'If you are not writing your own RenderBox subclass, then this is not\n'
+          'your fault. Contact support: https://github.com/flutter/flutter/issues/new?template=2_bug.yml',
+        )
+      ];
+
+      // Only perform the baseline checks after performLayout. Currently this
+      // method is also run in the RenderBox.size setter and after
+      // performResize, and it's not safe to access the baseline location in
+      // those two places since it may depend on the child layout.
```

Comment:
What do you mean by this comment? Isn't this only called from debugPaint?

Reply:
uhm I probably meant this should only be called in paint. Updated.

Reply:
Oh I think by "this method is also run in the ..." I was referring to other intrinsic checks not this method.